src/modules/job-exec/job-exec.c

```diff
@@ -729,6 +729,11 @@ static void jobinfo_start_continue (flux_future_t *f, void *arg)
         jobinfo_fatal_error (job, errno, "reading R: %s", error.text);
         goto done;
     }
+    if (job->multiuser
+        && !(job->J = strdup (jobinfo_kvs_lookup_get (f, "J")))) {
```

Comment:
I couldn't find anything that would indicate `strdup (NULL)` is required to return NULL, which I think is the assumption here.  Might want to handle `strdup()` and `jobinfo_kvs_lookup_get()` failures separately.

Reply:
Hm, there might be other problems in this function. Other fields like `jobspec` and `R` are not copied, but afaics the future containing these results is destroyed at function exit. I think so far no crashes because maybe the data is only accessed while the function is still running.

Probably the easiest thing to do here is to save the future until data isn't needed anymore.